ci-operator/step-registry/gather/alibabacloud-console/gather-alibabacloud-console-commands.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+
+
+if test -f "${KUBECONFIG}"
+then
+	oc --request-timeout=5s get nodes -o jsonpath --template '{range .items[*]}{.spec.providerID}{"\n"}{end}' | sed 's|.*/||' > "${TMPDIR}/node-provider-IDs.txt" &
+	wait "$!"
+
+	oc --request-timeout=5s -n openshift-machine-api get machines -o jsonpath --template '{range .items[*]}{.spec.providerID}{"\n"}{end}' | sed 's|.*/||' >> "${TMPDIR}/node-provider-IDs.txt" &
+	wait "$!"
+else
+	echo "No kubeconfig; skipping providerID extraction."
+fi
+
+if test -f "${SHARED_DIR}/alibaba-instance-ids.txt"
+then
+
+	export ALIBABA_CLOUD_CREDENTIALS_FILE="${SHARED_DIR}/alibabacreds.ini"
+	export ALIBABA_CLI_CREDENTIALS_FILE="${SHARED_DIR}/config"
+
+	pushd /tmp
+	wget https://aliyuncli.alicdn.com/aliyun-cli-linux-latest-amd64.tgz -O aliyun-cli.tgz
+	tar zxvf aliyun-cli.tgz
+	popd
+
+	/tmp/aliyun version
+
+	echo "Settting --config-path=${ALIBABA_CLI_CREDENTIALS_FILE} and --region=${LEASED_RESOURCE}"
+	/tmp/aliyun --config-path "${ALIBABA_CLI_CREDENTIALS_FILE}" configure set --region "${LEASED_RESOURCE}"
+
+
+	cat "${SHARED_DIR}/alibaba-instance-ids.txt" >> "${TMPDIR}/node-provider-IDs.txt"
```

Comment:
I think you want to decouple the `aliyun` setup from the ID collection.  Something like:

```sh
if test -f "${SHARED_DIR}/alibaba-instance-ids.txt"
then
  cat "${SHARED_DIR}/alibaba-instance-ids.txt" >> "${TMPDIR}/node-provider-IDs.txt"
fi

# If you really feel motivated, you could exit 0 here if there were no IDs in the assembled node-provider-IDs.txt , but I think that's unlikely enough that we can ignore it.  The existing AWS and GCP console gatherers don't bother.

pushd /tmp
# blah, blah, install aliyun
```